src/Http/Middleware/SessionCsrfProtectionMiddleware.php

```diff
@@ -268,4 +269,24 @@ protected function validateToken(ServerRequestInterface $request, Session $sessi
             'CSRF token from either the request body or request headers did not match or is missing.'
         ));
     }
+
+    /**
+     * Replace the token in the provided request.
+     *
+     * Replace the token in the session and request attribute. Replacing
+     * tokens is a good idea during privilege escalation or privilege reduction.
+     *
+     * @param \Cake\Http\ServerRequest $request The request to update
+     * @param string $key The session key/attribute to set.
+     * @return \Cake\Http\ServerRequest An updated request.
+     */
+    public static function replaceToken(ServerRequest $request, string $key = 'csrfToken'): ServerRequest
```

Comment:
Should we call this `updateToken()` since the user doesn't pass in a replacement?